pkg/cvo/availableupdates_test.go

```diff
@@ -0,0 +1,317 @@
+		Type:   configv1.RetrievedUpdates,
+		Status: configv1.ConditionTrue,
+	}
+
+	err := optr.syncAvailableUpdates(context.Background(), cvFixture)
+
+	if err != nil {
+		t.Fatalf("syncAvailableUpdates() unexpected error: %v", err)
+	}
+	if diff := cmp.Diff(expectedAvailableUpdates, optr.availableUpdates, availableUpdatesCmpOpts...); diff != "" {
+		t.Fatalf("available updates differ from expected:\n%s", diff)
+	}
+
+}
+
+func TestSyncAvailableUpdates_ConditionalUpdateRecommendedConditions(t *testing.T) {
+	testCases := []struct {
+		name                string
+		modifyOriginalState func(condition *metav1.Condition)
+		expectTimeChange    bool
+	}{
+		{
+			name:                "lastTransitionTime is not updated when nothing changes",
+			modifyOriginalState: func(condition *metav1.Condition) {},
+		},
+		{
+			name: "lastTransitionTime is not updated when changed but status is identical",
+			modifyOriginalState: func(condition *metav1.Condition) {
+				condition.Reason = "OldReason"
+				condition.Message = "This message should be changed to something else"
+			},
+		},
+		{
+			name: "lastTransitionTime is updated when status changes",
+			modifyOriginalState: func(condition *metav1.Condition) {
+				condition.Status = metav1.ConditionUnknown
+			},
+			expectTimeChange: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			conditionalUpdates, fakeOsus := osusWithSingleConditionalEdge("4.5.5", "4.5.6", promQL, evalToYes)
+			defer fakeOsus.Close()
+			availableUpdates, optr := newOperator(fakeOsus.URL, "4.5.5")
+			availableUpdates.ConditionalUpdates = conditionalUpdates
+			availableUpdates.Channel = cvFixture.Spec.Channel
+			availableUpdates.Condition = configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.RetrievedUpdates,
+				Status: configv1.ConditionTrue,
+			}
+			availableUpdates.ConditionalUpdates = conditionalUpdates
+			optr.availableUpdates = availableUpdates
+			expectedConditions := []metav1.Condition{{}}
+			conditionalUpdates[0].Conditions[0].DeepCopyInto(&expectedConditions[0])
+			tc.modifyOriginalState(&optr.availableUpdates.ConditionalUpdates[0].Conditions[0])
+
+			err := optr.syncAvailableUpdates(context.Background(), cvFixture)
+
+			if err != nil {
+				t.Fatalf("syncAvailableUpdates() unexpected error: %v", err)
+			}
+			if optr.availableUpdates == nil || len(optr.availableUpdates.ConditionalUpdates) == 0 {
+				t.Fatalf("syncAvailableUpdates() did not properly set available updates")
+			}
+			if diff := cmp.Diff(expectedConditions, optr.availableUpdates.ConditionalUpdates[0].Conditions, cmpopts.IgnoreTypes(time.Time{})); diff != "" {
+				t.Errorf("conditions on conditional updates differ from expected:\n%s", diff)
+			}
+			timeBefore := expectedConditions[0].LastTransitionTime
+			timeAfter := optr.availableUpdates.ConditionalUpdates[0].Conditions[0].LastTransitionTime
+			if tc.expectTimeChange == (timeBefore == timeAfter) {
```

Comment:
Nit: For some reason I had to bring out my rubber duck :duck: for this condition

But a simple alternative would just be:
```suggestion
			timeChange := !(timeBefore == timeAfter)
			if tc.expectTimeChange != timeChange {
```

Which isn't any better. So feel free to ignore this suggestion.

Reply:
No, this is a good point. This is not an uncommon smartass pattern in tests but you are right, why not simplify. I made it two ifs and got rid of the `not` kludge instead.